app/src/main/java/com/kodeco/android/coordplot/MainActivity.kt

```diff
@@ -46,10 +97,36 @@ fun PlotSurfacePreview() {
 
 @Composable
 fun Map(xPercent: Float, yPercent: Float, modifier: Modifier = Modifier) {
-    // TODO fill out the square map here.
-    //  Create a Box composable with a size of 300.dp
-    //  and then create an inner Box composable
-    //  with a clip shape of CircleShape.
+    val density = LocalDensity.current.density
+    var boxSize by remember { mutableStateOf(Size(0, 0)) }
+
+    Box(
+        modifier = modifier
+            .size(PLOT_SIZE.dp)
+            .aspectRatio(1f) // Ensure the plot is a square (1:1 aspect ratio)
+            .background(PLOT_BACKGROUND_COLOR)
+            .onGloballyPositioned { coordinates ->
+                boxSize = Size(
+                    ((coordinates.size.width / density).toInt()),
+                    ((coordinates.size.height / density).toInt())
+                )
+            }
+    ) {
+        val dotRadius = DOT_DIAMETER / 2
+        val offsetX = (xPercent * boxSize.width)
+        val offsetY = (yPercent * boxSize.height)
+
+        Box(
+            modifier = modifier
+                .offset(
+                    x = offsetX.dp - dotRadius.dp,
+                    y = offsetY.dp - dotRadius.dp
```

Comment:
And here, you can update this logic slightly.  Basically take the offset and subtract the dot radius first, _then_ convert the new value to dp:

```suggestion
                    x = (offsetX - dotRadius).dp,
                    y = (offsetY - dotRadius).dp
```